src/test/unit/controllers/AuthContollerTests.ts

```diff
@@ -0,0 +1,139 @@
+import * as AuthController from "../../../main/controllers/AuthControllers";
+import * as AuthService from "../../../main/services/AuthService";
+import { expect } from 'chai';
+import sinon from 'sinon';
+import express from "express";
+import { LoginRequest } from "../../../main/models/LoginRequest";
+import "express-session";
+declare module "express-session" {
+  interface SessionData {
+    token: string;
+  }
+}
+
+describe('AuthController', function () {
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  describe('getLoginForm', function () {
+    it('should render login form view', async () => {
+      const req = {};
+      const res = { render: sinon.spy() };
+
+      await AuthController.getLoginForm(req as express.Request, res as unknown as express.Response);
+
+      expect(res.render.calledOnce).to.be.true;
+    });
+  });
+
+  describe('postLoginForm', function () {
+    it('should post login form and redirect to logout form view', async () => {
+
+      const loginRequestObj: LoginRequest = {
+        email: "adam@random.com",
+        password: "pass123"
+      }
+
+      sinon.stub(AuthService, 'getTokenByloggingIn').resolves('12345');
+
+      const req = {
+        body: loginRequestObj,
+        session: { token: '12345' }
```

Comment:
don't set the token to that here. instead expect that the token has that value after calling the postLoginForm mtehod

Reply:
Sorry but I'm lost with this. If I don't define the session token this error occurs:
     TypeError: Cannot set properties of undefined (setting 'errormessage')

It returns the stub value but when we try to assign it to req.session.token it breaks.

I imaging we are setting the wrong variable though I don't know which one we need to set

Reply:
set the value as empty string or null or undefined then expect the value to be set to '12345' by the method call